src/sortStudents.ts

```diff
@@ -1,16 +1,180 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function isValuesEqual(arr: Student[], sortType: SortType): boolean {
+  for (let i = 0; i < arr.length - 1; i += 1) {
+    let currenValue = arr[i][sortType];
+    let nextValue = arr[i + 1][sortType];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    if (Array.isArray(currenValue) && Array.isArray(nextValue)) {
+      currenValue = currenValue.reduce(
+        (sum: number, current: number): number => {
+          return sum + current;
+        }, 0,
+      );
+
+      nextValue = nextValue.reduce(
+        (sum: number, current: number): number => {
+          return sum + current;
+        }, 0,
+      );
+    }
+
+    if (currenValue !== nextValue) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudets: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Age: {
+      const isEqual = isValuesEqual(students, sortBy);
+
+      if (isEqual) {
+        return copyStudets;
+      }
+
+      if (order === 'asc') {
+        return copyStudets.sort(
+          (a: Student, b: Student) => {
+            return a[SortType.Age] - b[SortType.Age];
+          },
+        );
+      }
+
+      if (order === 'desc') {
+        return copyStudets.sort(
+          (a: Student, b: Student) => {
+            return b[SortType.Age] - a[SortType.Age];
+          },
+        );
+      }
```

Comment:
you can reduce the code
```suggestion
        return copyStudets.sort(
          (a: Student, b: Student) => ({
            asc: a[SortType.Age] - b[SortType.Age],
            desc: b[SortType.Age] - a[SortType.Age]
          })[order],
        );
```

Reply:
fix everywhere